src/EPR.Calculator.API.UnitTests/Exporter/CalcResultsExporterTests.cs

```
@@ -0,0 +1,489 @@
+
+            //Act
+            var result = exporter.Export(results);
+
+            //Assert
+            Assert.IsTrue(result.Contains("5 Scheme set up cost Yearly Cost"));
+        }
+
+        [TestMethod]
+        public void Export_ShouldIncludeSummaryData_WhenNotNull()
+        { 
+            // Arrange
+            var results = CreateCalcResult();
+            var exporter = new CalcResultsExporter();
+
+            //Act
+            var result = exporter.Export(results);
+
+            //Assert
+
+            Assert.IsTrue(result.Contains("SummaryData"));
+        }
+
+        [TestMethod]
+        public void AppendDataIfNotNull_ShouldCallAppendMethod_WhenDataIsNotNull()
+        {
+            var csvContent = new StringBuilder();
+            var exporter = new CalcResultsExporter();
+            var data = new object();
+
+            exporter.AppendDataIfNotNull(data, (d, sb) => sb.Append("Data"), csvContent);
+
+            Assert.IsTrue(csvContent.ToString().Contains("Data"));
+        }
+
+        [TestMethod]
+        public void AppendDataIfNotNull_ShouldNotCallAppendMethod_WhenDataIsNull()
+        {
+            // Arrange
+            var csvContent = new StringBuilder();
+            var exporter = new CalcResultsExporter();
+            object? data = null;
+
+            // Act
+            exporter.AppendDataIfNotNull(data, (d, sb) => sb.Append("Data"), csvContent);
+
+            // Assert
+            Assert.IsFalse(csvContent.ToString().Contains("Data"));
+        }
+
+        [TestMethod]
+        public void Export_ShouldGenerateEmptyCsv_WhenNoData()
+        {
+            // Arrange
+            var results = new CalcResult
+            {
+                CalcResultLapcapData = null!,
+                CalcResultLateReportingTonnageData = null!,
+                CalcResultParameterOtherCost = null!
+            };
+            var exporter = new CalcResultsExporter();
+
+            // Act
+            if (results != null)
```

This test will silently pass if the results is null. Is it expected or do we need to have Assert outside if?